frontend/app_flowy/packages/flowy_infra/lib/theme.dart

```diff
@@ -1,267 +1,131 @@
-    }
-  }
-
-  ThemeData getThemeData(Locale locale) {
-    // Poppins and SF Mono are not well supported in some languages, so use the
-    // built-in font for the following languages.
-    final useBuiltInFontLanguages = [
-      const Locale('zh', 'CN'),
-      const Locale('zh', 'TW'),
-    ];
-    TextStyles textTheme;
-    if (useBuiltInFontLanguages.contains(locale)) {
-      textTheme = TextStyles(font: '', color: shader1);
-    } else {
-      textTheme = TextStyles(font: font, color: shader1);
-    }
-    return ThemeData(
-      brightness: brightness,
-      textTheme: textTheme.generateTextTheme(),
-      textSelectionTheme: TextSelectionThemeData(
-        cursorColor: main2,
-        selectionHandleColor: main2,
-      ),
-      primaryIconTheme: IconThemeData(color: hover),
-      iconTheme: IconThemeData(color: shader1),
-      scrollbarTheme: ScrollbarThemeData(
-        thumbColor: MaterialStateProperty.all(Colors.transparent),
-      ),
-      materialTapTargetSize: MaterialTapTargetSize.shrinkWrap,
-      canvasColor: shader6,
-      dividerColor: shader6,
-      hintColor: shader3,
-      disabledColor: shader4,
-      highlightColor: main1,
-      indicatorColor: main1,
-      toggleableActiveColor: main1,
-      colorScheme: ColorScheme(
-        brightness: brightness,
-        primary: main1,
-        onPrimary: _white,
-        primaryContainer: main2,
-        onPrimaryContainer: _white,
-        secondary: hover,
-        onSecondary: shader1,
-        secondaryContainer: selector,
-        onSecondaryContainer: shader1,
-        background: surface,
-        onBackground: shader1,
-        surface: surface,
-        onSurface: shader1,
-        onError: shader7,
-        error: red,
-        outline: shader4,
-        surfaceVariant: bg1,
-        shadow: shadow,
-      ),
-      extensions: [
-        AFThemeExtension(
-          warning: yellow,
-          success: green,
-          tint1: tint1,
-          tint2: tint2,
-          tint3: tint3,
-          tint4: tint4,
-          tint5: tint5,
-          tint6: tint6,
-          tint7: tint7,
-          tint8: tint8,
-          tint9: tint9,
-          greyHover: bg2,
-          greySelect: bg3,
-          lightGreyHover: shader6,
-          toggleOffFill: shader5,
-          code: textTheme.getFontStyle(fontFamily: monospaceFont),
-          callout: textTheme.getFontStyle(
-            fontSize: FontSizes.s11,
-            fontColor: shader3,
-          ),
-          caption: textTheme.getFontStyle(
-            fontSize: FontSizes.s11,
-            fontWeight: FontWeight.w400,
-            fontColor: shader3,
-          ),
-        )
-      ],
+  // metadata member
+  final FlowyColorScheme? lightTheme;
+  final FlowyColorScheme? darkTheme;
+  // static final Map<String, dynamic> _cachedJsonData = {};
+
+  const AppTheme({
+    this.lightTheme,
+    this.darkTheme,
+  });
+
+  factory AppTheme.fromName({required String themeName}) {
```

Comment:
Create a new factory method for loading JSON in the future.  Maybe `AppTheme.fromJson`

Reply:
Will this be a public factory? I was thinking that since the backend will only store the theme name, it will be the fromName factory that will determine whether to use a built-in theme or load from json.